### ISSUE TITLE
Consider adding LL_REVERSE to utlist.h

### ISSUE THREAD

**Issue by u-u-h** [troydhanson/uthash#197]
It's a common paradigm to use `LL_PREPEND` in a loop and a REVERSE step afterwards to avoid `LL_APPEND` in list construction. It would be nice to have that built-in in `utlist.h`

**Comment by Quuxplusone** [comment]
Yep, makes sense to me. Pull requests welcome; or if not an implementation, at least some sample test cases showing the desired usage would be nice.

Presumably `DL_REVERSE` and `CDL_REVERSE` should be added in the same PR.

**Comment by Quuxplusone** [comment]
Fixed by #278, 6d8573997c21f24c7e4ec9e48734b44f384170a1 — thanks @xunicatt!
